Assets/Scripts/UI/JoinLobbyButton.cs

```diff
@@ -0,0 +1,16 @@
+using Lobby;
+using UnityEngine;
+
+namespace UI
+{
+    public class JoinLobbyButton : MonoBehaviour, IButtonPressHandler
+    {
+        [SerializeField] private LobbyManager LobbyManager;
```

Comment:
Variable name being the same as your class name is going to lead to potential issues later on. Why not _lobbyManager, or something similar.

Reply:
@JudeWhite I've named a property the same as its type before without encountering problems before, so I'd be interested to know more about the potential issues. However, in this case we're breaking naming conventions and Rider would suggest the appropriate name. We'd have coding guidelines in place for this kind of stuff anyway, so it wouldn't be a matter of discussion in PRs.